server/container/guice/cassandra-rabbitmq-guice/src/main/java/org/apache/james/modules/TaskSerializationModule.java

```diff
@@ -110,6 +110,11 @@ protected void configure() {
         return CassandraMappingsSolveInconsistenciesTask.module(migration, dao);
     }
 
+    @ProvidesIntoSet
+    public TaskDTOModule<?, ?> clearMailQueueTask(MailQueueFactory<?> mailQueueFactory) {
+        return ClearMailQueueTaskDTO.module((MailQueueFactory<ManageableMailQueue>) mailQueueFactory);
```

Comment:
what make you think this assumption will stand?

Reply:
all the injection code is based on this assumption :/

Reply:
see for example https://github.com/linagora/james-project/blob/master/server/protocols/webadmin/webadmin-mailqueue/src/main/java/org/apache/james/webadmin/routes/MailQueueRoutes.java#L102

Reply:
This sucks.

One of the possible proposition would be not to propose Manageable part for a James-distribted MDA product, and I believe even if we do not do that move being flexible on this is a good thing for users assembling their own Guice products.

I'd rather create a separate issue about that 'the mailqueue is manageable' assumption. Maybe we need a clear API split?

Reply:
see https://issues.apache.org/jira/browse/JAMES-2926